### PR TITLE
Introduce inclusive naming check and update existing language

### PR DESCRIPTION
**Pull request by aaronshurley**
We are following the Inclusive Naming Initiative's guidance as that is what the CNCF supports.

### CODE REVIEW

**Comment by aaronshurley** [comment]
created as a draft until https://github.com/vmware-tanzu/carvel/pull/595 lands

File: pkg/vendir/fetch/image/imgpkg_test.go [hunk 1]
```diff
@@ -239,20 +239,20 @@ func requireImgpkgEnv(t *testing.T, expectedEnv, actualEnv []string) {
 	require.Equal(t, expectedEnv, filteredActualEnv)
 }
 
-type dummyCache struct {
+type sampleCache struct {
```
Comment (joaopapereira) [inline pkg/vendir/fetch/image/imgpkg_test.go]: After some soul searching this in fact looks much more like a fake [test double](https://en.wikipedia.org/wiki/Test_double) than really a dummy object test double.
```suggestion
type fakeCache struct {
```

**Review by joaopapereira — CHANGES_REQUESTED** [review]
I made some comments. Do you mind doing the proposed change?

**Comment by aaronshurley** [comment]
> I made some comments. Do you mind doing the proposed change?

Done. Re-submitted the PR.

**Review by joaopapereira — APPROVED** [review]
LGTM

**Pull request merged**